resdk/resources/utils.py

```diff
@@ -161,6 +161,14 @@ def get_samples(resource):
             samples.extend(get_samples(res))
         return samples
 
+    elif is_data(resource):
+        if not resource.sample:
+            raise TypeError(
+                "Resource should be sample, have `samples` query or be list of multiple resources."
```

Comment:
You should add this option to the error message - i.e. Data object with not empty `sample` property.

Also, as the same message is used few lines below, save it to the variable and use it instead.